src/Microsoft.ML.TensorFlow/TensorflowTransform.cs

```diff
@@ -178,33 +178,61 @@ private static VersionInfo GetVersionInfo()
         }
 
         /// <summary>
-        /// Creates <see cref="IDataTransform"/> using <see cref="TensorFlowTransform"/>.
-        /// This convenience method get the model file as input and loads the model internally.
-        /// If the model is already loaded please <see cref="TensorFlowTransform.Create(IHostEnvironment, IDataView, TensorFlowModelInfo, string[], string[])"/> to avoid reloading of model.
+        /// Transform for scoring Tensorflow models. Input data column names/types must exactly match
+        /// all model input names. Only the output columns specified will be generated.
+        /// If the model is already loaded please <see cref="TensorFlowTransform(IHostEnvironment, TensorFlowModelInfo, string, string)"/> to avoid reloading of model.
         /// </summary>
-        /// <param name="env">Host Environment.</param>
-        /// <param name="input">Input <see cref="IDataView"/>. This is the output from previous transform or loader.</param>
-        /// <param name="model">Path to the TensorFlow model. </param>
-        /// <param name="names">Name of the output column(s). Keep it same as in the Tensorflow model.</param>
-        /// <param name="source">Name of the input column(s). Keep it same as in the Tensorflow model.</param>
-        public static IDataTransform Create(IHostEnvironment env, IDataView input, string model, string[] names, string[] source)
+        /// <param name="env">The environment to use.</param>
+        /// <param name="modelFile">Model file path.</param>
+        /// <param name="inputColumn">The name of the input data column. Must match model input name.</param>
+        /// <param name="outputColumn">The output columns to generate. Names must match model specifications. Data types are inferred from model.</param>
+        public TensorFlowTransform(IHostEnvironment env, string modelFile, string inputColumn, string outputColumn)
```

Comment:
>TensorFlowTransform [](start = 15, length = 19)

The naming I find incomprehensible, similar to the ONNX thing. It is called a `Transform`, yet it appears to be a `Transformer`? Do we know what is going on here?

Reply:
It's a transformer now

---
In reply to: [245458201](https://github.com/dotnet/machinelearning/pull/2031#discussion_r245458201) [](ancestors = 245458201)